### PR TITLE
initialize directory while creating checkpoint file store

### LINKED ISSUE #41616: kubelet keeps reporting errors
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

**Kubernetes version** (use `kubectl version`):

**Environment**:
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
Using `hack/local-up-cluster.sh` to start a cluster with only one node and found that there are some error logs in kubelet as following:

```
E0217 11:15:39.175803   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:40.178541   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:41.181541   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:42.184452   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:43.187075   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:44.191545   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
E0217 11:15:45.194913   31536 docker_sandbox.go:370] Failed to list checkpoints: failed to list checkpoint store: open /var/lib/dockershim/sandbox: no such file or directory
```

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by freehan**
fixes: #41616 
ref: https://github.com/kubernetes/kubernetes/issues/41225

### CODE REVIEW

File: pkg/kubelet/dockershim/checkpoint_store.go [hunk 1]
```diff
@@ -57,6 +57,16 @@ type FileStore struct {
 	path string
 }
 
+func NewFileStore(path string) (CheckpointStore, error) {
+	if _, err := os.Stat(path); err != nil {
+		// if directory already exists, proceed
+		if err = os.MkdirAll(path, 0755); err != nil && !os.IsExist(err) {
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/checkpoint_store.go]: `MkdirAll` will not return exists error, right?

**Comment by yujuhong** [comment]
/approve

Looks good, pending addressing @Random-Liu's comment

File: pkg/kubelet/dockershim/checkpoint_store.go (hunk 1 above)
Comment (freehan) [inline pkg/kubelet/dockershim/checkpoint_store.go]: Looks like you are right
https://golang.org/pkg/os/#MkdirAll

**Comment by freehan** [comment]
fixed

File: pkg/kubelet/dockershim/checkpoint_store.go [hunk 2]
```diff
@@ -113,6 +117,17 @@ func (fstore *FileStore) getCheckpointPath(key string) string {
 	return filepath.Join(fstore.path, key)
 }
 
+// ensurePath creates input directory if it does not exist
+func ensurePath(path string) error {
+	if _, err := os.Stat(path); err != nil {
+		// MkdirAll returns nil if directory already exists
+		if err = os.MkdirAll(path, 0755); err != nil {
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/checkpoint_store.go]: you can just `return os.MkdirAll(path, 0755)`

**Comment by Random-Liu** [comment]
LGTM

**Comment by freehan** [comment]
adding label based on https://github.com/kubernetes/kubernetes/pull/41665#issuecomment-281232606

**Comment by freehan** [comment]
@k8s-bot cvm gce e2e test this
@k8s-bot gce etcd3 e2e test this

**Pull request merged**